pandas/tests/groupby/test_bin_groupby.py

```diff
@@ -20,7 +20,7 @@ def test_series_grouper():
     grouper = libreduction.SeriesGrouper(obj, np.mean, labels, 2)
     result, counts = grouper.get_result()
 
-    expected = np.array([obj[3:6].mean(), obj[6:].mean()])
+    expected = np.array([obj[3:6].mean(), obj[6:].mean()], dtype=object)
```

Comment:
these look like internal tests, but is anything user facing changed? (no looks like)? 

why are these now returning object? (e.g. can you just remove the cast_back above)?

Reply:
> (e.g. can you just remove the cast_back above)?

Figuring out how to do that is the next step.  ATM doing that breaks a handful of tests.  This PR doesn't change any user-facing behavior.

Reply:
follow-up with this is about ready